tests/containers/buildah.pm

```diff
@@ -0,0 +1,71 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Package: buildah
+# Summary: building OCI-compatible sle base images with buildah.
+# - install buildah
+# - create and run container
+# - build image with service and test connectivity
+# - cleanup system (images, containers)
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base qw(consoletest);
+use testapi;
+use utils;
+use containers::common;
+use containers::container_images;
+use containers::urls 'get_suse_container_urls';
+use version_utils qw(get_os_release check_os_release);
+
+sub run {
+    my ($running_version, $sp, $host_distri) = get_os_release;
+
+    install_docker_when_needed($host_distri);
```

Comment:
installing all together on the same instance : 
1. Not the same test case as install one on the clean host 
2. More error prone because you might get some conflicts on dependency level or during installation. while such errors should be catch , not sure if it make sense to block all testing when you can not install docker and podman on same machine. 
But generally I like idea of merging this very similar files . I would just switch approach from simply merging everything into one flow to something like this https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/main_common.pm#L1826 . where you still have single file ....

---------------------------------------------
at this point I realized that my suggestion will hit the wall which is called YAML_SCHEDULE :( there is still the way how to workaround it but I am not sure anymore if it really make sense . Let's discuss this better live ...

Reply:
Yes, it's not perfect. At first I wanted to give the variable from outside and call `buidalh.pm` 2 times, one with docker and other with podman as input variable, but I couldn't find a way to do so with yaml schedule. I thought of test_data, but that doesn't solve the problem either.

Reply:
The only workaround I can think of is to create a file somewhere `/tmp/docker_tested` at the end of the docker tests and then when the module is called again, it checks if that file exists and test podman if so...but that's very hacky  :D

Reply:
you could use testdata i guess. 
with testdata you can have lets say `runtime=docker` and then if `$testdata->{runtime} == 'docker' {} else {}`
But this seems more simple with a test variable

Reply:
> you could use testdata i guess.
> with testdata you can have lets say `runtime=docker` and then if `$testdata->{runtime} == 'docker' {} else {}`
> But this seems more simple with a test variable

this was exactly my idea , but looks like you getting into same trap as I did - do you know how to  bypass `$testdata`  into super cool and awesome feature called `YAML_SCHEDULE`  ?:)

Reply:
I know about test data in yaml schedules, but this defines the data for all the modules defined in that schedule. So all the modules get the same key/values... so you can't inject a variable to a specific module.

Reply:
> this was exactly my idea , but looks like you getting into same trap as I did - do you know how to bypass `$testdata` into super cool and awesome feature called `YAML_SCHEDULE` ?:)

What do you mean? This is actually possible: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/schedule/yast/autoyast/autoyast_mini.yaml#L34

and from the test module you can do `my $test_data = get_test_suite_data;`